frontend/src/app/admin/data.service.ts

```diff
@@ -12,23 +11,8 @@ import { environment } from '../../environments/environment'
 
 @Injectable()
 export class DataService {
-    private keys$: Observable<Key[]>;
```

Comment:
I believe this class is now unused. Please delete it and the import in `admin\admin.module.ts`.